test/controllers/home_controller_test.rb

```diff
@@ -0,0 +1,15 @@
+require 'test_helper'
+
+class HomeControllerTest < ActionDispatch::IntegrationTest
+  include Devise::Test::IntegrationHelpers
+
+  test 'should get homepage' do
```

Comment:
I think it'd be nice to take out the part where the user signs in. That could be tested separately inside `test/controllers/users/omniauth_callbacks_controller_test.rb`

So we'd have here:
```
# (I like to reference the HTTP method and the URL that's being tested, but this is just my style) 
test 'GET /home' do
  sign_in users(:schmidt) 

  get root_url

  assert_response :success
end
```

The tests for `users/sign_in`, `users/sign_up` and `users/sign_out` would go into the Omniauth controller.

Reply:
As we already discussed, I will move checking user's sign in/sign up/sign out with devise and google to system test.